src/System.Runtime.Numerics/tests/BigInteger/cast_from.cs

```diff
@@ -549,11 +549,28 @@ public static void RunDoubleExplicitCastFromBigIntegerTests()
             bigInteger += BigInteger.One;
             VerifyDoubleExplicitCastFromBigInteger(Double.MaxValue, bigInteger);
 
+            // Double Explicit Cast from BigInteger: Double.MinValue - 1
+            bigInteger = new BigInteger(Double.MinValue);
+            bigInteger -= BigInteger.One;
+            VerifyDoubleExplicitCastFromBigInteger(Double.MinValue, bigInteger);
+
             // Double Explicit Cast from BigInteger: Random value > Double.MaxValue
             bigInteger = GenerateRandomBigIntegerGreaterThan(Double.MaxValue, s_random);
             bigInteger *= 2;
             VerifyDoubleExplicitCastFromBigInteger(Double.PositiveInfinity, bigInteger);
 
+            // Double Explicit Cast from BigInteger: Random value < -Double.MaxValue
+            VerifyDoubleExplicitCastFromBigInteger(Double.NegativeInfinity, -bigInteger);
+
+            // Double Explicit Cast from BigInteger: very large values (more than Int32.MaxValue bits) should be infinity
+            bigInteger = BigInteger.One;
+            for (int i = 0; i < 5; ++i)
```

Comment:
The tests are taking quite a bit longer on my machine, a little over twice as long (16 sec vs 7 sec originally). We may want to add a quicker repro test and put this version into a separate method that has an [OuterLoop] attribute on it. That will let us run this version in our nightly "full" test runs, but only run a minimal version of it on every pull request.